src/item.cpp

```diff
@@ -5778,22 +5778,68 @@ bool item::process_food( player * /*carrier*/, const tripoint &pos )
     if( item_tags.count( "FROZEN" ) > 0 && item_counter > 500 && type->comestible->parasites > 0 ) {
         item_tags.insert( "NO_PARASITES" );
     }
+
+    // minimum is 0 - takes into account that process() takes --1 counter per turn regardless
+    // div by 5 means every 5 degrees of difference equals 1 point of ratio
     /* cache g->get_temperature( item location ). It is used a minimum of 3 times, no reason to recalculate. */
-    const auto item_local_temp = g->get_temperature( pos );
-    unsigned int diff_freeze = abs( item_local_temp - FREEZING_TEMPERATURE );
-    diff_freeze = diff_freeze < 1 ? 1 : diff_freeze;
-    diff_freeze = diff_freeze > 10 ? 10 : diff_freeze;
-
-    unsigned int diff_cold = abs( item_local_temp - FRIDGE_TEMPERATURE );
-    diff_cold = diff_cold < 1 ? 1 : diff_cold;
-    diff_cold = diff_cold > 10 ? 10 : diff_cold;
-    // environment temperature applies COLD/FROZEN flags to food
-    if( item_local_temp <= FRIDGE_TEMPERATURE ) {
-        g->m.apply_in_fridge( *this, item_local_temp );
-    } else if ( item_tags.count( "FROZEN" ) > 0 && item_counter > diff_freeze ) {
-        item_counter -= diff_freeze;
-    } else if( item_tags.count( "COLD" ) > 0 && item_counter > diff_cold ) {
-        item_counter -= diff_cold;
+    const auto temp = g->get_temperature( pos );
+    unsigned int diff_freeze = abs( temp - FREEZING_TEMPERATURE ) / 5 ;
+    diff_freeze = clamp( diff_freeze, static_cast<unsigned int>(0) , static_cast<unsigned int>(3) ); //effective 1-4
+
+    // div by 5 means every 5 degrees of difference equals 1 point of ratio
+    unsigned int diff_cold = abs( temp - FRIDGE_TEMPERATURE ) / 5;
+    diff_cold = clamp( diff_cold, static_cast<unsigned int>(0), static_cast<unsigned int>(3) ); //effective 1-4
+
+    // exclusions for items being actively cooled to be affected by outside temperatures
+    //const optional_vpart_position vp = g->m.veh_at( pos );
+    const optional_vpart_position vp = g->m.veh_at( pos );
+    bool in_active_cooler = false;
+    bool fridge_here = false;
+    bool freezer_here = false;
+    if( vp ) {
+        fridge_here = vp->vehicle().has_part( pos, "FRIDGE", true );
+        freezer_here = vp->vehicle().has_part( pos, "FREEZER", true );
+        if(fridge_here) { add_msg( "Fridge" ); } else {add_msg( "NOT Fridge" );}
+        if(freezer_here) { add_msg( "Freezer" ); } else {add_msg( "NOT Freezer" );}
+    }
+
+    if( vp && ( fridge_here || freezer_here ) ) {
+        std::vector<vehicle_part *> parts = vp->vehicle().get_parts( pos, "FRIDGE", true );
+        if( parts.empty() ) { 
+            parts = vp->vehicle().get_parts( pos, "FREEZER", true );
+        }
+        if( parts.empty() ) { add_msg( "PARTS EMPTY" ); }
+        vehicle_part *part = parts.front();
+        vehicle_stack vit = vp->vehicle().get_items( vp->vehicle().index_of_part( part ) );
+        if( vit.empty() ) { add_msg( "ITEMS EMPTY" ); }
+        for( size_t iter = 0; iter < vit.size(); iter++ ) {
```

Comment:
I think the vehicle_stack is too complicated for what you want do here. You have a vehicle_part and an item, you just want to know if they is in the vehicle part. You don't want to adjust the item lists, you just want to confirm the location of a item you already have. So try this:
```
diff --git a/src/vehicle.h b/src/vehicle.h
index 3e653a4258..2b9f9c8413 100644
--- a/src/vehicle.h
+++ b/src/vehicle.h
@@ -134,6 +134,9 @@ struct vehicle_part {
         /** Amount of fuel, charges or ammunition currently contained by a part */
         long ammo_remaining() const;
 
+        /** Check if the inventory contains an item */
+        bool has_item( const item *it ) const;
+
         /**
          * Set fuel, charges or ammunition for this part removing any existing ammo
          * @param ammo specific type of ammo (must be compatible with vehicle part)
diff --git a/src/vehicle_part.cpp b/src/vehicle_part.cpp
index 6bc8e01f4f..2c3c915b4a 100644
--- a/src/vehicle_part.cpp
+++ b/src/vehicle_part.cpp
@@ -158,6 +158,15 @@ bool vehicle_part::is_available( const bool carried ) const
     return !is_unavailable( carried );
 }
 
+bool vehicle_part::has_item( const item *it ) const
+{
+    for( auto iter: items ) {
+        if( &iter == it ) {
+            return true;
+        }
+    }
+    return false;
+}
 itype_id vehicle_part::ammo_current() const
 {
     if( is_battery() ) {
```
Assuming you can get that to work, you can replace the code after this
```
    if( vp && ( fridge_here || freezer_here ) ) {
```
with
```
        in_active_cooler = vp->has_item( this );
```

Reply:
`vp` is not a `vehicle part` class but `optional_vpart_position` class so this will not work that simply, but thanks for the `has_item` idea, it might help a bit